src/app/input-bar/input-bar.component.ts

```diff
@@ -18,25 +19,167 @@ export class InputBarComponent implements OnInit {
-    this.ccdService.updateRegex(options.regex);
-    this.ccdService.updateGroupingKey(options.group);
+    this.ccdService.updateFilters(options.filters);
+    this.ccdService.updateGroupingKey(options.groupBy);
 
-    // we delete any options that are not filter by metadata keys so that
-    // only filter keys are left for the filter pipe to manage.
-    delete options.regex;
-    delete options.group;
-    this.ccdService.updateFilters(options);
+    // We need to scan filters for regex so we can highlight matches
+    const newRegexArr: RegExp[] = [];
+    options.filters.forEach((filter: FilterOption<any>) => {
+      if (filter.filterKey === FilterKeys.Regex && !filter.negation) {
+        newRegexArr.push(new RegExp(`(${filter.inputString})`, 'g'));
+      }
+    });
+    this.ccdService.updateRegex(newRegexArr);
 
   }
 
-  parseInput(input: string) {
-    try {
-      return JSON.parse(input);
-    } catch (e) {
-      // By deault we assume user is just inputting a regex
-      return {regex: input};
+  private parseInput(input: string) {
+    const filters: FilterOption<any>[] = [];
+    let groupBy: GroupByKeys = GroupByKeys.None;
+
+    if (!input) {
+      return {
+        filters,
+        groupBy
+      };
+    }
+    const splitInput = input.trim().split(' ');
+
+    if (splitInput.length === 1) {
+      const newFilter = this.parseArgument(splitInput[0], true);
+      if (newFilter) {
+        filters.push(newFilter);
+      }
+    } else {
+      for (let i = 0; i < splitInput.length; i++) {
+        if (splitInput[i] !== '{') {
+          const newFilter = this.parseArgument(splitInput[i], false);
+          if (newFilter.validInput) {
+            filters.push(newFilter);
+          }
+        } else if (i + 2 >= splitInput.length || splitInput[i + 2] !== '}') {
+          // To group we require the syntax '{ grouping_key }'
+          console.error('Invalid input, to group links make sure there are ' +
+              'spaces between the brackets and the grouping key.');
+          break;
+        } else {
+          for (const key in GroupByKeys) {
+            if (GroupByKeys[key] === splitInput[i + 1]) {
+              groupBy = GroupByKeys[key];
+              break;
+            }
+          }
+          i += 2;
+        }
+      }
     }
+
+    return {
+      filters,
+      groupBy
+    };
   }
+
+  /**
+   * Parses a full filter argument, parses it and constructs a FilterOption
+   * object filled with the relevant data.
+   *
+   * @param { string } str The full filter argument as a string
+   * @param { boolean } onlyArg Whether it's the only argument and we should
+   * accept it as a regex filter if there is no modifier
+   * @returns { FilterOption<any> } The constructed FilterOption object
+   */
+  private parseArgument(str: string, onlyArg: boolean): FilterOption<any> {
+    // By default we assume a regex unless we find a modifier
+    const filterOption = {
+      filterKey: undefined,
```

Comment:
Let's use None enum instead of undefined

Reply:
Good catch!